arviz/plots/tsplot.py

```diff
@@ -0,0 +1,142 @@
+"""Plot timeseries data."""
+import numpy as np
+
+from ..sel_utils import xarray_var_iter
+from ..rcparams import rcParams
+from .plot_utils import get_plotting_function
+
+
+def plot_ts(
+    idata,
+    y=None,
+    x=None,
+    components=None,
+    holdout=-1,
+    num_samples=100,
+    backend=None,
+    backend_kwargs=None,
+    textsize=None,
+    figsize=None,
+    axes=None,
+    show=None,
+):
+    """Plot timeseries and it's components.
+
+    Parameters
+    ----------
+    idata : az.InferenceData object, Optional
+        Optional only if y is Sequence
+    y : str, variable name from observed_data
+    x : str, Optional
+        If none, coords name of y (y should be DataArray).
+    components : list, variables from posterior, Optional
+        Shape of the components must be like (chain, draw, *)
+    holdout : int, Optional
+    num_samples : int, Optional, Default 100
+    backend : str, Optional
+        Select plotting backend {"matplotlib","bokeh"}. Default "matplotlib".
+    backend_kwargs : dict, optional
+    figsize : tuple, optional
+        Figure size. If None it will be defined automatically.
+    textsize : float, optional
+        Text size scaling factor for labels, titles and lines. If None it will be
+        autoscaled based on figsize.
+
+
+    Returns
+    -------
+    axes: matplotlib axes or bokeh figures.
+    """
+    if isinstance(y, str):
+        y_hat = idata.posterior_predictive[y]
+        y = idata.observed_data[y]
+
+    if isinstance(x, str):
+        x = idata.constant_data[x]
+    else:
+        x = y.coords[y.dims[0]]
+
+    components_plotters = []
+    comp_uncertainty_plotters = []
+    total_samples = y_hat.sizes["chain"] * y_hat.sizes["draw"]
+    pp_sample_ix = np.random.choice(total_samples, size=num_samples, replace=False)
+
+    if components is not None:
+        for component in components:
+            component_xr = idata.posterior[component]
+
+            component_mean = component_xr.mean(("chain", "draw"))
+            components_plotters += list(
+                xarray_var_iter(
+                    component_mean,
+                    skip_dims=set(component_mean.dims),
+                    combined=True,
+                )
+            )
+
+            components_satcked = component_xr[..., -holdout:].stack(sample=("chain", "draw"))[
```

Comment:
this assumes that the dim on which holdout should be done is the last one, which may not be the case. It should be `.isel(dim_name=slice(None, holdout)`

Also, the name of the stacked dimension should be `__sample__` instead to avoid name clashes. see #1613 for context